src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -4803,60 +4806,75 @@ namespace winrt::TerminalApp::implementation
     //     color of the titlebar)
     // Arguments:
     // - requestedTheme: this should be the currently active Theme for the app
+    // - activeBorderColor: the pane's border color for the application when it is active
+    // - inactiveBorderColor: the pane's border color for the application when it is inactive
+    // - broadcastBorderColor: the pane's border color for the application when it is broadcast
     // Return Value:
     // - <none>
-    void TerminalPage::_updatePaneResources(const winrt::Windows::UI::Xaml::ElementTheme& requestedTheme)
+    void TerminalPage::_updatePaneResources(const winrt::Windows::UI::Xaml::ElementTheme& requestedTheme, const winrt::Microsoft::Terminal::Settings::Model::ThemeColor& activeBorderColor, const winrt::Microsoft::Terminal::Settings::Model::ThemeColor& inactiveBorderColor, const winrt::Microsoft::Terminal::Settings::Model::ThemeColor& broadcastBorderColor)
     {
         const auto res = Application::Current().Resources();
         const auto accentColorKey = winrt::box_value(L"SystemAccentColor");
+        auto activeBrushColor = Colors::Black(), inactiveBrushColor = Colors::Black(), broadcastBrushColor = Colors::Black();
         if (res.HasKey(accentColorKey))
         {
             const auto colorFromResources = ThemeLookup(res, requestedTheme, accentColorKey);
             // If SystemAccentColor is _not_ a Color for some reason, use
             // Transparent as the color, so we don't do this process again on
             // the next pane (by leaving s_focusedBorderBrush nullptr)
-            auto actualColor = winrt::unbox_value_or<Color>(colorFromResources, Colors::Black());
-            _paneResources.focusedBorderBrush = SolidColorBrush(actualColor);
-        }
-        else
+            activeBrushColor = winrt::unbox_value_or<Color>(colorFromResources, Colors::Black());
+        }
+        const auto copyTerminalColorObjToUIColorObj = [](const winrt::Microsoft::Terminal::Settings::Model::ThemeColor& colorToCopy) {
+            auto rawColor = colorToCopy.Color();
+            return winrt::Windows::UI::Color{
+                rawColor.A,
+                rawColor.R,
+                rawColor.G,
+                rawColor.B
+            };
+        };
```

Comment:
The `ThemeColor` may also be the accent color or terminal background. You're going to have to check the `ThemeColor::ColorType` to know which one it is, and if it's one of those special colors, you'll have to deduce it. Looks like the accent color is already retrieved above, so that's good. "Terminal background" on the other hand loks like it needs more work.

Reply:
Reworked so terminalBackground and accentColor now work, thanks